### PR TITLE
Address ropensci review

### LINKED ISSUE #200: `use_tic()`: Ensure that a DESCRIPTION file is present
Otherwise, macro functions will error during the CI run. See https://github.com/ropensci/software-review/issues/305#issuecomment-565823673.

- [ ] Assert a DESCRIPTION file in `use_tic()`. Create one if missing? Repo names with `-` are problematic.

- [ ] Mention (more clearly) that build dependencies are installed from packages listed in the DESCRIPTION file

### PR DESCRIPTION
**Pull request by pat-s**
- fixes #200 

https://github.com/ropensci/software-review/issues/305#issuecomment-565804724

- [x] mention the difference between .com and .org -> _new vignette "org-vs-com")_

- [x]  indeed, perhaps the default R_TRAVIS should be .com given the current changes to travis? -> _changed in {travis}_

- [x] During travis::travis_sync() the API token is pasted in the console and (although deleted form history) is still visible in the console. Could this be avoided in any way? -> _done when refactoring {travis}_

- [x] return early if `do_*` and `step_*` functions are called interactively

- [x] error and suggest to call `use_tic()` first when running `dsl_init()` in a repo without a `tic.R`

- [x] Make it possible to pass the `endpoint` arg from {travis} funs to `use_tic()`

- README updates

- all examples are not `\dontrun` because `dsl_init()` only works within a project

---

Because of 

https://github.com/ropenscilabs/tic/blob/834b30f0c63b004015154bfb94fd8c30b18cb5c8/R/stage.R#L35-L45

I had to move {usethis} from suggests to imports. Otherwise {usethis} is missing as it is not yet installed in the "install" step as a dep when being called in `tic::prepare_stages()`.

I would like to avoid this actually but we are using quite a bit of {usethis} internally all over the place and maybe we should just have it in imports.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -5,7 +5,7 @@ Title: Tasks Integrating Continuously: CI-Agnostic Workflow
 Version: 0.3.0.9002
 Authors@R: 
     c(person(given = "Kirill",
-             family = "M\u00fcller",
+             family = "Müller",
```
Comment (krlmlr) [inline DESCRIPTION]:
```suggestion
             family = "M\u00fcller",
```

File: DESCRIPTION [hunk 2]
```diff
@@ -48,24 +48,27 @@ Imports:
     R6,
     remotes,
     rlang (>= 0.4.1.9000),
+    usethis,
```
Comment (krlmlr) [inline DESCRIPTION]: I'm not sure we should import this.

File: R/dsl-storage.R [hunk 3]
```diff
@@ -118,10 +118,15 @@ dsl_load <- function(path = "tic.R", force = FALSE, quiet = FALSE) {
 #' @rdname dsl_get
 #' @export
 dsl_init <- function(quiet = FALSE) {
+  path <- file.exists(paste0(usethis::proj_get(), "/tic.R"))
```
Comment (krlmlr) [inline R/dsl-storage.R]: `dsl_init()` is designed to always work, and is used in the examples. Can we leave it as is?

**Comment by krlmlr** [comment]
I resolved conflicts and tentatively undid the code changes. We can discuss.

This is what a call to `do_bookdown()` looks like now:

``` r
tic::do_bookdown()
#> Using TRAVIS_DEPLOY_KEY env var as the private key name for SSH deployment.
#> ✓ Creating a blank tic stage configuration
#> ℹ tic configuration
#> ── install ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── stage ──
#> ▶ step_install_deps(repos = repo_default())
#> ── deploy ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── stage ──
#> ▶ step_build_bookdown()
```

<sup>Created on 2020-01-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

We can add a note that points to `?tic::use_tic` here.

Testing for interactive code is brittle: users might want to run `tic::run_all_stages()` in their interactive session.

I also don't think that {tic} should unconditionally require a `DESCRIPTION`. Rather it's the responsibility of the steps to check that a file is available, and err appropriately. Perhaps in their "prepare" step? (The "troubleshooting" section needs to mention `prepare_all_stages()` I think.)

Perhaps we should link more prominently to the "troubleshooting" section, to assist first time users? Or move "troubleshooting: local runs" to the "Get started" document?

**Comment by krlmlr** [comment]
Regarding `proj_get()`: Again, I think tic can expect to find a `tic.R` and everything else in its working directory, at least on the CI system. What motivated the change here?

**Comment by krlmlr** [comment]
Currently:

``` r
tic::do_bookdown()
#> Using TRAVIS_DEPLOY_KEY env var as the private key name for SSH deployment.
#> ✓ Creating a clean tic stage configuration
#> ℹ See `?tic::dsl_get` for details
#> 
#> ── tic configuration ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
#> ── install ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── stage ──
#> ▶ step_install_deps(repos = repo_default())
#> ── deploy ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── stage ──
#> ▶ step_build_bookdown()
```

<sup>Created on 2020-01-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

The help page explains what's going on, pointing to the `tic.R` file and the vignette.

Checking `DESCRIPTION` in steps that need it. Added `prepare_all_stages()` to troubleshooting. Added one more link to troubleshooting.

**Comment by krlmlr** [comment]
@pat-s: I can't request your review. Could you please take a look?

**Comment by pat-s** [comment]
> Regarding proj_get(): Again, I think tic can expect to find a tic.R and everything else in its working directory, at least on the CI system. What motivated the change here?

The motivation came from the ropensci review where Anna tried calling some tic functions on a new repo without a tic.R file - i.e. no run of `use_tic()` as the first action.

> This is what a call to do_bookdown() looks like now:

Should we expand the message something like: "We detected you are running XYZ interactively, which is not supported. See ?tic::dsl_get for details."?

I think we have room to improve the documentation on how to use {tic} interactively:

- not calling steps/macros standalone
- use `tic::run_stage("<stage>")` or `tic::run_all_stages()`

This would have helped Anna in her review I assume. We should also prominently link such a resource in interactive calls to macros/steps like the one above. What do you think?

File: R/local.R [hunk 4]
```diff
@@ -4,17 +4,22 @@ LocalCI <- R6Class(
...
     get_slug = function() {
-      remote <- gh::gh_tree_remote()
-      paste0(remote$username, "/", remote$repo)
+      tryCatch(
+        {
+          remote <- gh::gh_tree_remote()
+          paste0(remote$username, "/", remote$repo)
+        },
+        error = ""
+      )
```
Comment (pat-s) [inline R/local.R]: What motivated this tryCatch?

File: R/local.R [hunk 5]
```diff
@@ -23,7 +28,7 @@ LocalCI <- R6Class(
       NULL
     },
     get_commit = function() {
-      git2r::revparse_single(revision = "HEAD")$sha
+      tryCatch(git2r::revparse_single(revision = "HEAD")$sha, error = "")
```
Comment (pat-s) [inline R/local.R]: What motivated this tryCatch?

File: R/steps-install.R [hunk 6]
```diff
@@ -9,6 +9,9 @@ InstallDeps <- R6Class(
     },
 
     prepare = function() {
+      if (!file.exists("DESCRIPTION")) {
+        stopc("No DESCRIPTION file found. The step_install_deps step and the do_package_checks macro are only available for packages.")
```
Comment (pat-s) [inline R/steps-install.R]: I recently like to format error messages as

```r
cli_alert_danger("<extended descriptive message>")
stopc("No DESCRIPTION file found.")
```

The advantage is that you can wrap the extended description and also use cli formatting in it. The plain message in the stop call is then just to state the main reason of the error.

Comment (krlmlr) [reply R/steps-install.R]: rlang conditions are better here, see `?rlang::cnd_body` . Here I'm using the pattern you're suggesting, for consistency.

Comment (pat-s) [reply R/steps-install.R]: Thanks, looks promising.

**Pull request merged**